src/ngx-gallery-scroll-overview/ngx-gallery-scroll-overview.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, Input, AfterViewInit } from '@angular/core';
+import { NgxGalleryPreviewComponent } from '../ngx-gallery-preview.component';
+
+@Component({
+    selector: 'ngx-gallery-scroll-overview',
+    templateUrl: './ngx-gallery-scroll-overview.component.html',
+    styleUrls: ['./ngx-gallery-scroll-overview.component.scss']
+})
+export class NgxGalleryScrollOverviewComponent implements AfterViewInit{
+    private beforeZoomLeft: number;
+    private beforeZoomTop: number;
+    private initialLeft: number;
+    private initialTop: number;
+    private SCALE_FACTOR = 8;
+
+    @Input() gallery: NgxGalleryPreviewComponent;
+
+    inBounds = true;
+    edge = {
+        top: true,
+        bottom: true,
+        left: true,
+        right: true
+    };
+
+    constructor() {
+
+    }
+
+    public updateDetailZoom() {
+        var topScaled = (this.initialTop - this.gallery.positionTop) / this.SCALE_FACTOR;
+        var leftScaled = (this.initialLeft - this.gallery.positionLeft) / this.SCALE_FACTOR;
+        var transformString = 'translate(' + leftScaled + 'px,' + topScaled + 'px)';
+        document.getElementById('zoomContainer').style.transform = transformString;
+    }
+
+    public updatePreviewScales() {
+        var img = document.getElementsByClassName('ngx-gallery-fullsize');
+        var width = img[0].clientWidth;
+        var height = img[0].clientHeight;
+
+        // set the background image of the preview Container which contains the Scaled preview Image
+        document.getElementById('previewContainer').style.backgroundImage = "url('"+ img[0].getAttribute('src') +"')";
+
+        // scale the preview image
+        var widthScaled = width / this.SCALE_FACTOR;
```

Comment:
Define those variables at the beginning of the component and use it in the template via ngStyle